Gems/AtomLyIntegration/CommonFeatures/Code/Source/PostProcess/EditorModeFeedback/EditorEditorModeFeedbackSystemComponent.cpp

```diff
@@ -0,0 +1,268 @@
+            AZ::Interface<EditorModeFeedbackInterface>::Unregister(this);
+            AzToolsFramework::ViewportEditorModeNotificationsBus::Handler::BusDisconnect();
+            AzToolsFramework::Components::EditorComponentBase::Deactivate();
+        }
+
+        bool EditorEditorModeFeedbackSystemComponent::IsEnabled() const
+        {
+            return m_enabled;
+        }
+
+        void EditorEditorModeFeedbackSystemComponent::RegisterDrawableComponent(
+            EntityComponentIdPair entityComponentId, const MeshFeatureProcessorInterface::MeshHandle& meshHandle)
+        {
+            // Overwrite any existing mesh handle for this component-entity id
+            auto& meshHandleDrawPackets =
+                m_entityComponentMeshHandleDrawPackets[entityComponentId.GetEntityId()][entityComponentId.GetComponentId()];
+            meshHandleDrawPackets.m_meshHandle = &meshHandle;
+
+            // The same component can call RegisterDrawableEntity multiple times in order to update its model asset so always
+            // clear any existing draw packets for the component upon registration
+            meshHandleDrawPackets.m_meshDrawPackets.clear();
+        }
+
+        void EditorEditorModeFeedbackSystemComponent::OnEditorModeActivated(
+            [[maybe_unused]] const AzToolsFramework::ViewportEditorModesInterface& editorModeState,
+            AzToolsFramework::ViewportEditorMode mode)
+        {
+            // Purge the draw packets for all registered 
+            if (mode == AzToolsFramework::ViewportEditorMode::Focus)
+            {
+                for (auto& [entityId, componentMeshHandleDrawPackets] : m_entityComponentMeshHandleDrawPackets)
+                {
+                    for (auto& [componentid, meshHandleDrawPackets] : componentMeshHandleDrawPackets)
+                    {
+                        meshHandleDrawPackets.m_meshDrawPackets.clear();
+                    }
+                }
+
+                m_enabled = true;
+            }
+        }
+
+        void EditorEditorModeFeedbackSystemComponent::OnEditorModeDeactivated(
+            [[maybe_unused]] const AzToolsFramework::ViewportEditorModesInterface& editorModeState,
+            AzToolsFramework::ViewportEditorMode mode)
+        {
+            if (mode == AzToolsFramework::ViewportEditorMode::Focus)
+            {
+                m_enabled = false;
+            }
+        }
+
+        void EditorEditorModeFeedbackSystemComponent::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+        {
+            if (!m_enabled)
+            {
+                return;
+            }
+
+            const auto focusModeInterface = AZ::Interface<AzToolsFramework::FocusModeInterface>::Get();
+            if (!focusModeInterface)
+            {
+                return;
+            }
+
+            // TODO: see if there is a more reliable method of creating the required resources once the depender systems are initialized
```

Comment:
nit: Likely better to just have this be an internal task and remove the TODO from the source (or add a comment here in the PR rather than checking it in)

Reply:
Good catch! Removed 👍